### PR TITLE
Make the account field optional and let the user specify what is included in the URI

### PR DESCRIPTION
**Pull request by kylecbrodie**
I've made the account field optional because it isn't required in the [defacto standard](https://github.com/google/google-authenticator/wiki/Key-Uri-Format).

I also added some logic to generate URIs that only contains the following information:
otpauth://{otp_type}/{issuer:account or just issuer}?secret={secret}&issuer={issuer}

The current version of the Google Authenticator app considers QR codes with more parameters invalid.

### CODE REVIEW

**Comment by steveherrin** [comment]
Not requiring an account seems reasonable. If you want to split that into a separate PR, I'd be happy to merge that.

The standard (the same one you linked to) says "algorithm", "digits", and "period" are ignored by the Google authenticator implementations. So if it's not accepting those (even to ignore them), it sounds like a problem with their implementation ignoring their own standard. Have you raised the issue with them?

That aside, two thoughts:
1. This should be a parameter to the `get_uri` method. I can generate urls with both the minimal set of parameters and the full set. There's no need to pick one or the other initially.
2. I'd say having a parameter for which optional parameters to include in the uri would be a better solution than a boolean flag. That would always let the user get a uri that's compatible with whatever quirks their authenticator app has. For one that's compatible with google authenticator, you'd pass in something like `uri_params=['secret', 'issuer']`.

**Comment by kylecbrodie** [comment]
I haven't raised the issue with them because I don't know where to report it. Their open-source repositories: [google/google-authenticator](https://github.com/google/google-authenticator) and [google/google-authenticator-android](https://github.com/google/google-authenticator-android) both say they are outdated codebases and that patches aren't likely to be incorporated upstream.

Your idea of replacing the flag with an optional parameter where you can list what should be included in the URI was great. I've included it in the latest set of changes.

File: spookyotp/otp.py [hunk 1]
```diff
@@ -70,47 +70,59 @@ def _get_algorithm(algorithm_name):
...
         from the filename.
         """
-        qr_code = self.get_qr_code()
+        qr_code = self.get_qr_code(uri_params)
         qr_code.save(filename)
 
     @classmethod
-    def _get_uri(cls, secret, issuer, account,
-                 n_digits, algorithm, **other_params):
+    def _get_uri(cls, secret, issuer, account=None,
```
Comment (steveherrin) [inline spookyotp/otp.py]: What might make more sense is for this to be `_get_uri(cls, **kwargs)`, and then the logic would be:
1. Combine the account and the issuer, if an account was specified
2. Start forming the uri, including the protocol and the account/issuer
3. Append the secret
4. Append `'{0}={1}'.format(key, quote(value))` for everything else in kwargs

And then the public `get_uri` method would handle figuring out which parameters to pass into `_get_uri`.

Something roughly like

```
def get_uri(self, uri_params=None):
    params = {secret: self._secret,
                     issuer: self._issuer}
    for (param, attr) in self._all_parameters:
        if uri_params is None or param in uri_params:
            params[param] = getattr(self, attr)
    return self._get_uri(**params)
```

Does that sound reasonable? That simplifies `_get_uri` by letting it treat most optional parameters equally. The public methods handle translating what the user wants into arguments to `_get_uri`.

**Comment by steveherrin** [comment]
Hey @kylecbrodie, I'm really sorry for the long delay in replying. I think I waiting for the tests to pass, and then I got distracted and didn't follow-up.

I put in changes that hopefully accomplish what you were looking for, in version 1.1.0. Please open a new issue or submit a new PR if those aren't satisfactory.

Thanks for your help bringing this to my attention and proposing this fix.

**Pull request closed without merging**